katz/src/main/kotlin/katz/instances/IOSemigroup.kt

```diff
@@ -0,0 +1,11 @@
+package katz
+
+class IOSemigroup<A>(val SM: Semigroup<A>) : Semigroup<HK<IO.F, A>> {
```

Comment:
The companion object has invoke overloaded to provide a default value. n this case IO acts as a wrapper, and the SemiGroup depends mostly on the contents. We could have separate invokes for common types that are not globally registered like numbers and lists.